src/Plugins/LevelDBStore/IO/Data/LevelDB/DB.cs

```diff
@@ -122,5 +125,19 @@ public void Write(WriteOptions options, WriteBatch write_batch)
             Native.leveldb_write(Handle, options.Handle, write_batch.Handle, out var error);
             NativeHelper.CheckError(error);
         }
+
+        public IEnumerable<KeyValuePair<byte[], byte[]>> GetAll(Snapshot? snapshot = null)
+        {
+            using var options = new ReadOptions();
+            if (snapshot != null) options.Snapshot = snapshot;
+
+            using var iterator = NewIterator(options);
+            iterator.SeekToFirst();
+            while (iterator.Valid())
```

Comment:
Why is this `while`, and below is `for`. Might it be better to keep a consistent style?

Reply:
He copied mine. but `for` would be better.

Reply:
will be fixed in another pr.